client/asset/interface.go

```diff
@@ -198,6 +198,8 @@ type AuditInfo interface {
 	Expiration() time.Time
 	// Coin is the coin that contains the contract.
 	Coin() Coin
+	// Contract is the contract script.
+	Contract() dex.Bytes
```

Comment:
Ah. I see. Will remove.

Reply:
I now see where it is used outside of tests. In `Redeem()`, to validate the secret in `asset.Redemption.Secret`:
https://github.com/decred/dcrdex/blob/7e386d1a5d9f18ba9c93c541f347300501e3f6c2/client/asset/dcr/dcr.go#L1116-L1129

Previously, the contract was retrieved from the coin:
https://github.com/decred/dcrdex/blob/aa65868f238a08395947ec6590455104f8636dcc/client/asset/dcr/dcr.go#L1085

I can either pass the `Secret` in `asset.Redemption` since the caller does have that info, or keep it in `AuditInfo` to make `AuditInfo` the source of truth. I'm for the latter unless you think otherwise.

Reply:
It's also possible to store the contract in the implementation-specific `auditInfo` type and later access it after casting `r.Spends` (type asset.AuditInfo) to `*auditInfo` in `cinfo, ok := r.Spends.(*auditInfo)`.

But something I'd ultimately like to see is not requiring the type casting to succeed before performing redemption. The 
methods defined in the asset.AuditInfo interface should be sufficient to retrieve data needed to process the redemption request. For starters, I'm using the `r.Spends.Contract()` method to retrieve the contract rather than `cinfo.contract`.

Now that I think of it, doing away with that type-casting requirement is trivial and I'll just do it. It should not be a requirement that a `*auditInfo` be the underlying type for all `asset.Redemption.Spends` (type `asset.AuditInfo`).